mongodb/resource_automation_agent.go

```diff
@@ -42,39 +44,67 @@ func resourceMdbAutomationAgentCreate(data *schema.ResourceData, meta interface{
 	automationConfig := types.ReadAutomationAgentConfig(automation)
 
 	// create a SSH connection to the remote host
-	client, err := NewSSHClient(providerConfig, conn)
+	sshClient, err := NewSSHClient(providerConfig, conn)
 	if err != nil {
 		return fmt.Errorf("could not create a SSH client: %v", err)
 	}
 
 	// attempt to create directories if not already present
 	cmd := fmt.Sprintf("mkdir -p %s %s", automationConfig.WorkDir, automationConfig.LogPath)
-	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
-
+	ssh.PanicOnError(sshClient.RunCommand(conn.SudoPrefix(cmd)))
 	cmd = fmt.Sprintf("cd %s", automationConfig.WorkDir)
-	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
+	ssh.PanicOnError(sshClient.RunCommand(conn.SudoPrefix(cmd)))
 
 	// Set correct permissions for directories
 	cmd = fmt.Sprintf("chown `whoami` %s %s", automationConfig.WorkDir, automationConfig.LogPath)
-	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
+	ssh.PanicOnError(sshClient.RunCommand(conn.SudoPrefix(cmd)))
 
 	// download the automation agent binary on the remote host
 	filename := fmt.Sprintf("mongodb-mms-automation-agent-%s.linux_x86_64.tar.gz", automationConfig.Version)
 	cmd = fmt.Sprintf("curl -O \"%s/download/agent/automation/%s\"", automationConfig.MMSBaseURL, filename)
-	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
+	ssh.PanicOnError(sshClient.RunCommand(conn.SudoPrefix(cmd)))
 
 	// unpack the binary
 	cmd = fmt.Sprintf("tar -C %s -xvzf %s --strip 1", automationConfig.WorkDir, filename)
-	ssh.PanicOnError(client.RunCommand(cmd))
+	ssh.PanicOnError(sshClient.RunCommand(cmd))
 	log.Printf("[DEBUG] unpacked the binary in: %s", automationConfig.WorkDir)
 
-	// TODO: get the API key and projectID from client
+	// initialize a client without auth
+	apiURL := fmt.Sprintf("http://127.0.0.1:%d", automationConfig.OpsManagerPort)
+	resolver := httpclient.NewURLResolverWithPrefix(apiURL, opsmanager.PublicAPIPrefix)
+	withResolver := opsmanager.WithResolver(resolver)
+	withHTTPClient := opsmanager.WithHTTPClient(httpclient.NewClient())
+	omAPIClientNoAuth := opsmanager.NewClient(withResolver, withHTTPClient)
+
+	// create the first user
+	user := opsmanager.User{Username: "firstuser", Password: "password", FirstName: "first", LastName: "last"}
```

Comment:
note for a further PR: let's store the `username` in the OpsManager schema and use it for all three fields:
- Username
- Firstname
- Lastname